addons/pos_restaurant/static/src/js/floors.js

```diff
@@ -6,73 +6,52 @@ var pos_env = require('point_of_sale.env');
 const { Gui } = require('point_of_sale.Gui');
 const Registries = require('point_of_sale.Registries');
 
-// At POS Startup, load the floors, and add them to the pos model
-models.load_models({
-    model: 'restaurant.floor',
-    loaded: function(self,floors){
-        self.floors = floors;
-        self.floors_by_id = {};
-        for (var i = 0; i < floors.length; i++) {
-            floors[i].tables = [];
-            self.floors_by_id[floors[i].id] = floors[i];
-        }
-
-        // Make sure they display in the correct order
-        self.floors = self.floors.sort(function(a,b){ return a.sequence - b.sequence; });
-
-        // Ignore floorplan features if no floor specified.
-        self.config.iface_floorplan = !!self.floors.length;
-    },
-});
-
-// At POS Startup, after the floors are loaded, load the tables, and associate
-// them with their floor.
-models.load_models({
-    model: 'restaurant.table',
-    loaded: function(self,tables){
-        self.tables_by_id = {};
-        for (var i = 0; i < tables.length; i++) {
-            self.tables_by_id[tables[i].id] = tables[i];
-            var floor = self.floors_by_id[tables[i].floor_id[0]];
-            if (floor) {
-                floor.tables.push(tables[i]);
-                tables[i].floor = floor;
-            }
-        }
-    },
-});
 
 // New orders are now associated with the current table, if any.
 Registries.PosModelRegistry.extend(models.Order, (Order) => {
 
 class PosRestaurantOrder extends Order {
     constructor(obj, options) {
         super(...arguments);
-        if (!this.table && !options.json) {
-            this.table = this.pos.table;
+        if (this.pos.config.module_pos_restaurant) {
```

Comment:
there are several of this:
```js
        if (this.pos.config.module_pos_restaurant) {
            if (this.pos.config.iface_floorplan
```

maybe you can just combine them:

```js
if (this.pos.config.module_pos_restaurant && this.pos.config.iface_floorplan)
```

Reply:
No because there are code that happen in basic pos_restaurant and not essentially with `iface_floorplan` like `this.customer_count = this.customer_count || 1;`